### PR TITLE
Upgrade Kafka to 3.1.0

### LINKED ISSUE #52: Update to latest Kafka Connect API version to 3.x
Kafka dependency is at 1.x - update to the most recent 3.x version (3.1.0 at the time of writing).

### PR DESCRIPTION
**Pull request by emlaver**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [x] Added tests for code changes _or_ test/build only changes
- [x] Updated the change log file (`CHANGES.md`|`CHANGELOG.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description
- Upgrade Kafka Connect to 3.1.0
- Update Scala to 2.12 since it's the minimum version for [Kafka 3.1.0 download](https://kafka.apache.org/downloads).
- Upgrade logging dependency to slf4j

fixes #52

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # UNRELEASED
 - [UPGRADED] Upgraded Gradle distribution from 4.5.1 to 7.4
+- [UPGRADED] Upgraded Kafka Connect API to 3.1.0 and Scala version to 2.12.
```
Comment (ricellis) [inline CHANGES.md:3]:
```suggestion
- [UPGRADED] Upgraded Kafka Connect API to 3.1.0
```

File: src/main/java/com/ibm/cloudant/kafka/connect/CloudantSourceTask.java [hunk 2]
```diff
@@ -210,7 +211,7 @@ public void stop() {
             try {
                 Thread.sleep(SHUTDOWN_DELAY_MILLISEC);
             } catch (InterruptedException e) {
-                LOG.error(e);
+                LOG.error(e.getMessage());
```
Comment (ricellis) [inline src/main/java/com/ibm/cloudant/kafka/connect/CloudantSourceTask.java:214]: MIght be worth including the throwable so a full stack trace can be logged if desired.
```suggestion
                LOG.error(e.getMessage(), e);
```

**Review by ricellis — APPROVED** [review]
+1, but a couple of nits and copyright dates should probably be freshened.

**Review by vmatyusGitHub — APPROVED** [review]
LGTM :+1:

**Pull request merged**